### DISCUSSION TITLE
Not all reservations visible in the UI

### DISCUSSION THREAD

**Discussion by ybensoussan** [Azure/ipam#177] [General]
Hi, 

I am currently setting up some automation for our landingzones where I'm using this solution to determine the VNETs to provision. 
To do this I am using an admin SP calling the API (enige app registration). 
This works wonderfully and I get my reservations set up nicely to use later on in the provisioning of the vnets.

However, it seems that the reservations are only "visible" to the SP I created them with. (If I do an API call with the SP token I get a list of the reservations created by that user, but not any others).

This behavior is the same in the UI, I created 1 reservation with a token of an admin user using the API. This reservation is visible in the UI. But the reservations created with the admin SP are not visible.

Am I missing some permissions or is this intended behaviour?

Sincerely,
Yonathan

**Comment by DCMattyG** [comment]
Hi @ybensoussan good morning!

The expected behavior of Reservations is as follows:
-**Admin** users of Azure IPAM will be able to see **all** reservations
-**Non-Admin** users of Azure IPAM will be able to see **only their own** reservations

One other thing to note is that within the UI, by default it will only show **active** Reservations, but you can switch this by switching between **Showing Active** and **Showing All** from the drop-down in the upper right corner like this:

![image](https://github.com/Azure/ipam/assets/12955732/44b85a5f-494f-4333-aa73-9caeedd7b934)

Please let me know if this information helps, otherwise I'm more than happy to diver deeper into this with you and get to the bottom of the problem!

**Comment by ybensoussan** [comment]
Hey @DCMattyG ,

Thanks for the explanation, but it does not explain the behavior I'm seeing on my end.
Here's some more details. 

![image](https://github.com/Azure/ipam/assets/5622780/13aca131-4de6-4161-bef3-144fbcf3418a)

OK so here's my reservations window.
I am logged in as 

![image](https://github.com/Azure/ipam/assets/5622780/4f020481-ebd6-4723-8f6e-3df6ee424cfe)

But then if I run my playbook (ansible) I get a reservations that I created in my pipeline:

```
ok: [localhost] => {
    "msg": {
        "changed": false,
        "connection": "close",
        "content_length": "2304",
        "content_type": "application/json",
        "cookies": {
            "ARRAffinity": "8f78b3466a2e63914acc9f8042d2b59b3c1fb87f8aa26a97c5901d27812e606a",
            "ARRAffinitySameSite": "8f78b3466a2e63914acc9f8042d2b59b3c1fb87f8aa26a97c5901d27812e606a"
        },
        "cookies_string": "ARRAffinity=8f78b3466a2e63914acc9f8042d2b59b3c1fb87f8aa26a97c5901d27812e606a; ARRAffinitySameSite=8f78b3466a2e63914acc9f8042d2b59b3c1fb87f8aa26a97c5901d27812e606a",
        "date": "Fri, 29 Sep 2023 15:11:34 GMT",
        "elapsed": 0,
        "failed": false,
        "json": [
            {
                "block": "***",
                "cidr": "10.240.255.0/24",
                "createdBy": "spn:eda7a10a-c2b4-4766-b172-d833ebe628c8",
                "createdOn": 1695925433.926315,
                "desc": "vnet reservation for testing",
                "id": "9WgbbkLHCoUFFHrWqssy6m",
                "settledBy": null,
                "settledOn": null,
                "space": "***",
                "status": "wait",
                "tag": {
                    "X-IPAM-RES-ID": "9WgbbkLHCoUFFHrWqssy6m"
                }
            },
            {
                "block": "***",
                "cidr": "10.240.254.0/24",
                "createdBy": "spn:eda7a10a-c2b4-4766-b172-d833ebe628c8",
                "createdOn": 1695926471.4207027,
                "desc": "vnet reservation for testing",
                "id": "JWh4Cujau28TjkbB87Vkfr",
                "settledBy": null,
                "settledOn": null,
                "space": "***",
                "status": "wait",
                "tag": {
                    "X-IPAM-RES-ID": "JWh4Cujau28TjkbB87Vkfr"
                }
            },
            {
                "block": "***",
                "cidr": "10.240.253.224/27",
                "createdBy": "spn:eda7a10a-c2b4-4766-b172-d833ebe628c8",
                "createdOn": 1695926609.550023,
                "desc": "vnet reservation for testing",
                "id": "JVPNZZScydjuExG9Lt3Yyu",
                "settledBy": null,
                "settledOn": null,
                "space": "***"
                "status": "wait",
                "tag": {
                    "X-IPAM-RES-ID": "JVPNZZScydjuExG9Lt3Yyu"
                }
            },
            {
                "block": "****",
                "cidr": "10.240.253.192/27",
                "createdBy": "spn:eda7a10a-c2b4-4766-b172-d833ebe628c8",
                "createdOn": 1695974695.311092,
                "desc": "vnet reservation for trivial_with_network prd",
                "id": "7S23XkvHR35GWDBAgMYZ3r",
                "settledBy": null,
                "settledOn": null,`
```

After I created another reservation using postman and a token from the SP, that one DOES show up in the reservations window...

The automation is calling a private DNS entry, with postman I created the reservations against a public hostname. Could that be an issue?

**Comment by ybensoussan** [comment]
oh gawd... it's always DNS....
Just answered my own question.... we recently redeployed the solution, but didn't take down the test setup and didn't change the DNS entry to point to the new one..........
Closing this, thanks for your time @DCMattyG !

**Reply by DCMattyG** [reply]
Hey @ybensoussan, I'm always happy to be a therapist when DNS is being a bully, haha!

![image](https://github.com/Azure/ipam/assets/12955732/aec07a3d-d5b2-4513-a4e7-9098848f41c4)

Happy to help, even if it's just a shoulder to cry on 🤣
